prow/test/integration/fakegerritserver/fakegerritserver.go

```diff
@@ -98,21 +121,172 @@ func defaultHandler() func(*http.Request) (interface{}, int, error) {
 	}
 }
 
-func changesHandler() func(*http.Request) (interface{}, int, error) {
+// GetBranch
+func projectHandler(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
+	return func(r *http.Request) (interface{}, int, error) {
+		vars := mux.Vars(r)
+		projectName := vars["project-name"]
+		branchID := vars["branch-id"]
+		if res := fgc.GetBranch(projectName, branchID); res != nil {
+			content, err := json.Marshal(res)
+			if err != nil {
+				return "", http.StatusInternalServerError, err
+			}
+			return string(content), http.StatusOK, nil
+		}
+		return "", http.StatusNotFound, nil
+	}
+}
+
+// Admin endpoint to add a change to the Fake Gerrit Server
+func addHandler(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
+	return func(r *http.Request) (interface{}, int, error) {
+		change := gerrit.ChangeInfo{}
+		err := unmarshal(r, &change)
+		if err != nil {
+			logrus.Infof("Error unmarshling: %v", err)
+			return "", http.StatusInternalServerError, err
+		}
+		fgc.AddChange("fakegerritserver", &change)
+
+		logrus.Infof("The Entire Fgc: %v", fgc)
+		return "", http.StatusOK, nil
+	}
+}
+
+// Admin endpoint to reset the Fake Gerrit Server
+func resetHandler(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
+	return func(r *http.Request) (interface{}, int, error) {
+		fgc.Reset()
+		return "", http.StatusOK, nil
+	}
+}
+
+// Handles ListChangeComments
+func handleGetComments(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
 	return func(r *http.Request) (interface{}, int, error) {
 		logrus.Infof("Serving: %s, %s", r.URL.Path, r.Method)
 		vars := mux.Vars(r)
 		id := vars["change-id"]
-		if id == "1" {
-			content, err := json.Marshal(gerrit.ChangeInfo{
-				ChangeID: "1",
-			})
-			logrus.Debugf("JSON: %v", content)
+		comments := fgc.GetComments(id)
+		if comments == nil {
+			return "", http.StatusNotFound, nil
+		}
+		content, err := json.Marshal(comments)
+		if err != nil {
+			return "", http.StatusInternalServerError, err
+		}
+		return string(content), http.StatusOK, nil
+	}
+}
+
+func processQueryString(query string) string {
+	return strings.TrimPrefix(query, "project:")
+}
+
+// Handles QueryChanges
+func handleChangeQuery(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
```

Comment:
Should this be named `handleQueryChanges` instead? Comment above is out of sync with this function name, which hurts readability.

Reply:
Done

Reply:
Done